cloud-nio/cloud-nio-impl-drs/src/test/scala/cloud/nio/impl/drs/DrsCloudNioFileProviderSpec.scala

```diff
@@ -38,35 +37,26 @@ class DrsCloudNioFileProviderSpec extends AnyFlatSpecLike with CromwellTimeoutSp
     fileSystemProvider.isFatal(new RuntimeException) should be(false)
     fileSystemProvider.isTransient(new RuntimeException) should be(false)
     fileSystemProvider.getScheme should be("drs")
-    fileSystemProvider.getHost("drs://dg.123/abc") should be("dg.123")
-    fileSystemProvider.getHost("drs://dg.example.com/abc") should be("dg.example.com")
-  }
-
-  it should "be able to get the hostname from variously formatted DRS URIs" in {
-    val config = ConfigFactory.parseString(
-      """martha.url = "https://from.config"
-        |access-token-acceptable-ttl = 1 minute
-        |""".stripMargin
+    val exampleUris = List(
+      "drs://dg.123/abc",
+      "drs://dg.example.com/abc",
+      "drs://drs.data.humancellatlas.org/8aca942c-17f7-4e34-b8fd-3c12e50f9291?version=2019-07-04T151444.185805Z",
+      "drs://jade.datarepo-dev.broadinstitute.org" +
+        "/v1_93dc1e76-8f1c-4949-8f9b-07a087f3ce7b_8b07563a-542f-4b5c-9e00-e8fe6b1861de",
+      "drs://dg.4503/fc046e84-6cf9-43a3-99cc-ffa2964b88cb",
+      // Example non-W3C/IETF URIs provided by
+      // https://docs.google.com/document/d/1Wf4enSGOEXD5_AE-uzLoYqjIp5MnePbZ6kYTVFp1WoM/edit
+      "drs://dg.4DFC:0027045b-9ed6-45af-a68e-f55037b5184c",
+      "drs://dg.4503:dg.4503/fc046e84-6cf9-43a3-99cc-ffa2964b88cb",
+      "drs://dg.ANV0:dg.ANV0/0db6577e-57bd-48a1-93c6-327c292bcb6b",
+      "drs://dg.F82A1A:ed6be7ab-068e-46c8-824a-f39cfbb885cc",
```

Comment:
how about `drs://${randomCharGenerator.next(1000)}` as an exciting additional test case?

Reply:
Heh. We could-- but in general as these new URIs have appeared in the wild, I've tried to stop using random strings. Despite appearances, these are all real working<sup>*</sup> examples from the doc linked just above.

<sub>
<sup>*</sup> They will be working once these PRs are merged. We can't test them yet though until we implement some tech-debt called "RAS based testing".
</sub>

Reply:
@cjllanwarne @kshakir I know I'm late to the party here, but I want to weigh in strongly against randomness in this kind of test. Use of random inputs requires a deliberate practice of capturing failure cases and addressing them specifically, sometimes codifying them into their own non-random regression tests. Without that, it's likely that we'd just re-run the test, see it pass, 🤷 , and move on.